core/src/main/java/hudson/model/TopLevelItemDescriptor.java

```diff
@@ -286,4 +286,8 @@ public static ExtensionList<TopLevelItemDescriptor> all() {
         return Items.all();
     }
 
+    @Restricted(NoExternalUse.class)
+    public static FormValidation doCheckDisplayNameOrNull(@AncestorInPath AbstractProject<?, ?> project, @QueryParameter String value) {
```

Comment:
You can't use AbstractProject here. This must be TopLevelItem
`@AncestorInPath TopLevelItem item`